DeepSpeech.py

```diff
@@ -1150,6 +1164,28 @@ def _next_epoch(self):
         # Indicates, if there were 'new' epoch(s) provided
         result = False
 
+        # Make sure that early stop is enabled and validation part is enabled
+        if FLAGS.early_stop is True and FLAGS.validation_step > 0:
+
+            # Check for early stopping condition
+            if(len(self._dev_losses) >= FLAGS.earlystop_nsteps):
+
+                # Calculate the mean of losses for past epochs
+                mean_loss = np.mean(self._dev_losses[-FLAGS.earlystop_nsteps:-2])
+                # Calculate the standard deviation for losses from validation part in the past epochs
+                std_loss = np.std(self._dev_losses[-FLAGS.earlystop_nsteps:-2])
```

Comment:
its for picking up the latest error and comparing it with previous n_step losses.